Client/Frontend/UserScript/UserScript.js

```diff
@@ -0,0 +1,67 @@
+//
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ + * License, v. 2.0. If a copy of the MPL was not distributed with this
+ + * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+var noImageModeEnabled = false;
+var observers = new Array();
+var _firefox_NoImageMode = {
+    setNoImageMode: function(enabled) {
+        noImageModeEnabled = enabled;
+        for (var i=0; i<observers.length; i++) {
+            var observer = observers.shift();
+            observer.disconnect();
+        }
+        if (enabled) {
+            _firefox_NoImageMode.setStyle();
+            var bodys = document.getElementsByTagName("body");
+            for (var i=0; i<bodys.length; i++) {
+                _firefox_NoImageMode.blockImgs(bodys[i], noImageModeEnabled);
+                _firefox_NoImageMode.elemListener(bodys[i], noImageModeEnabled);
+            }
+        }
+    },
+
+    setStyle: function() {
+        var no_image_css_id = "iBrowser_no_image_css";
+        var no_image_css = '*{background-image:none !important;}img,iframe{visibility:hidden !important;}';
+        var newCss = document.getElementById(no_image_css_id);
+        if(newCss == undefined){
+            var cssStyle = document.createElement('style');
+            cssStyle.type = 'text/css';
+            cssStyle.id = no_image_css_id;
+            if (cssStyle.styleSheet) {
+                cssStyle.styleSheet.cssText = no_image_css;
+            } else {
+                cssStyle.appendChild(document.createTextNode(no_image_css));
+            }
+            document.documentElement.appendChild(cssStyle);
+        } else {
+            newCss.innerHTML = no_image_css;
+        }
+    },
+
+    blockImgs: function(elem, enabled) {
+        var imgs = document.getElementsByTagName("IMG");
+        for (var i=0; i<imgs.length;i++) {
+            var parent = imgs[i].parentNode;
+            parent.removeChild(imgs[i]);
+        }
+    },
+
+    elemListener: function(target, enabled){
+        var observer = new MutationObserver(function(mutations) {
+            mutations.forEach(function(mutation) {
+                _firefox_NoImageMode.blockImgs(mutation.addedNodes, enabled);
+            });
+        });
+        // configuration of the observer:
+        var config = { childList: true, subtree: true };
+        // pass in the target node, as well as the observer options
+        observer.observe(target, config);
+        observers.push(observer);
+    },
+}
+
+document.addEventListener("DOMContentLoaded", function(event) {
+    _firefox_NoImageMode.setNoImageMode(noImageModeEnabled);
+});
```

Comment:
All this code is visible to every webpage. It's an attack surface which we need to proactively minimize.

At minimum, you need to wrap this in a immediately executing closure, and then only attach the `setNoImageMode` function to the `__firefox__` object.

```
(function() {
  if (!window.__firefox__) {
    window.__firefox__ = {};
  }
  var observers = [];
  var isEnabled = false;

  function initializeStyleSheet()
  function blockImages(elem, enabled) {}
  function watchForImages(elem, enabled) {}

  window.__firefox__.setNoImageMode = function(enabled) {}

  window.addListener("DOMContentLoaded", function (event) {});
})();
```

This is the same as other tab helper `js` files.